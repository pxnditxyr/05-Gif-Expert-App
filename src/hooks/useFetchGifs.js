import { useState, useEffect } from 'react';
import { getFetchGifs } from '../helpers/getFetchGifs';

export const useFetchGifs = ( category ) => {

	const [ state, setState ] = useState({
		data: [],
		loading: true,
	});

	useEffect( () => {

		getFetchGifs( category )
			.then( gifs => {

				setTimeout( () => {
					setState({
						data: gifs,
						loading: false,
					});
				}, 3000);

			});

	}, [ category ]);

	return state;
};
