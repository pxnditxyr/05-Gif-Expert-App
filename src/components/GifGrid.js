// import React, { useState, useEffect } from 'react';
import React from 'react';
import PropTypes from 'prop-types';
import { useFetchGifs } from '../hooks/useFetchGifs';
// import { GifGridItem } from './GifGridItem';
// import { getGifs } from '../helpers/getGifs';


export const GifGrid = ({ category }) => {


	const { data, loading } = useFetchGifs();
	console.log( data );
	console.log( loading );
	// const [ images, setImages ] = useState( [] );
	// useEffect( () => {
	// 	getGifs( category )
	// 		.then( imgs => {
	// 			setImages( imgs );
	// 		});
	// }, [ category ])


	return (
		<>
			<h2> { category } </h2>
			{ loading ? 'Loading...' : 'Data charge done' }
			{/* <div className="card-grid"> */}
			{/* 	{ */}
			{/* 		images.map( img => ( */}
			{/* 			<GifGridItem */}
			{/* 				key={ img.id } */}
			{/* 				{ ...img  } */}
			{/* 			/> */}
			{/* 		)) */}
			{/* 	} */}
			{/* </div> */}
		</>
	);
};

GifGrid.prototype = {
	category: PropTypes.string.isRequired,
}
GifGrid.defaultProps = {
	category: "Nuevo"
}
