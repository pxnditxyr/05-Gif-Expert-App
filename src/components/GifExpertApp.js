import React, { useState } from 'react';
import { AddCategory } from './AddCategory';
import { GifGrid } from './GifGrid';

export const GifExpertApp = () => {

	// const categories = [ "Naruto", "Bleach", "One Piece" ];
	//const [ categories, setCategories ] = useState( ["Naruto", "Bleach", "One Piece", "KissxSis"] );
	const [ categories, setCategories ] = useState( ["Naruto"] );


	// const handleAdd = () => setCategories( [...categories, "KissxSis"] )

	return (

		<>
			<h2> GifExpertApp XD </h2>
			<AddCategory setCategories={ setCategories } />
			<hr />
			{/* <button onClick={ handleAdd }> Add </button> */}
			<ol>
				{
					categories.map( category =>
						<GifGrid
							key={ category }
							category={ category }
						/>
						//{/* <li key={ category }> { category } </li> */}
					)
				}
			</ol>
		</>
	);
};
