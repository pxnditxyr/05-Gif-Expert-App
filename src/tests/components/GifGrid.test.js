import React from 'react';
import { shallow } from 'enzyme';
import { GifGrid } from '../../components/GifGrid';
import { useFetchGifs } from '../../hooks/useFetchGifs';


jest.mock( "../../hooks/useFetchGifs" );

describe( "Test's on GifGrid component", () => {

	const category = "Kiss x Sis";
	// let wrapper;
	// beforeEach( () => {
	// 	wrapper = shallow( <GifGrid category={ category }/> );
	// });
	test( "Should show component correctly", () => {
		useFetchGifs.mockReturnValue({
			data: [],
			loading: true,
		});

		const wrapper = shallow( <GifGrid category={ category }/> );
		expect( wrapper ).toMatchSnapshot();
	});

	test( "Should show items when image is loaded", () => {

		const gifs = [{
			id: 'new',
			url: 'http://localhost/any/thing.jpg',
			title: 'Any Thing',
		}];

		useFetchGifs.mockReturnValue({
			data: gifs,
			loading: false,
		});
		const wrapper = shallow( <GifGrid category={ category }/> );
		expect( wrapper ).toMatchSnapshot();
	});


	test( "Must show items when image is loaded", () => {

		const gifs = [{
			id: 'new',
			url: 'http://localhost/any/thing.jpg',
			title: 'Any Thing',
		}];

		useFetchGifs.mockReturnValue({
			data: gifs,
			loading: false,
		});
		const wrapper = shallow( <GifGrid category={ category }/> );
		expect( wrapper.find( 'p' ).exists() ).toBe( false );
		expect( wrapper.find( 'GifGridItem' ).length ).toBe( gifs.length );
	});

});
