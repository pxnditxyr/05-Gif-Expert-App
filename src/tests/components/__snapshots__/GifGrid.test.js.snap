// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test's on GifGrid component Should show component correctly 1`] = `
<Fragment>
  <h2
    className="animate__animated animate__fadeIn"
  >
     
    Kiss x Sis
     
  </h2>
  <p
    className="animate__animated animate__flash"
  >
     Loading... 
  </p>
  <div
    className="card-grid"
  />
</Fragment>
`;

exports[`Test's on GifGrid component Should show items when image is loaded 1`] = `
<Fragment>
  <h2
    className="animate__animated animate__fadeIn"
  >
     
    Kiss x Sis
     
  </h2>
  <div
    className="card-grid"
  >
    <GifGridItem
      id="new"
      key="new"
      title="Any Thing"
      url="http://localhost/any/thing.jpg"
    />
  </div>
</Fragment>
`;
